File: pytools/data/data.code.lua.server/AckChatSendOk.lua
module("AckChatSendOk", package.seeall)


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckChatSendOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.channel = 0
	self.uid = 0
	self.uname = 0
	self.content = 0
end


function encode(self)
	pack = packet.new()
	pack:writeU8(self.channel)
	pack:writeU32(self.uid)
	pack:writeString(self.uname)
	pack:writeString(self.content)
	return pack:encode(Msg.P_ACK_CHAT_SEND_OK)
end


function setChannel(self, channel)
	self.channel = channel
end

function setUid(self, uid)
	self.uid = uid
end

function setUname(self, uname)
	self.uname = uname
end

function setContent(self, content)
	self.content = content
end

File: pytools/examples/golang/src/server/proto/ack_chat_send_ok.go
package proto

import (
	"packet"
)

type AckChatSendOk struct {
	channel                  uint8
	uid                      uint32
	uname                    string
	content                  string
}

func (this *AckChatSendOk) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteUint8(this.channel)
	pack.WriteUint32(this.uid)
	pack.WriteString(this.uname)
	pack.WriteString(this.content)

	return pack.Encode(P_ACK_CHAT_SEND_OK)
}

func (this *AckChatSendOk) SetChannel(channel uint8) {
	this.channel = channel
}

func (this *AckChatSendOk) SetUid(uid uint32) {
	this.uid = uid
}

func (this *AckChatSendOk) SetUname(uname string) {
	this.uname = uname
}

func (this *AckChatSendOk) SetContent(content string) {
	this.content = content
}
